src/node/cli.ts

```diff
@@ -106,6 +107,10 @@ async function resolveOptions(mode: string) {
     argv.root = path.isAbsolute(argv.root) ? argv.root : path.resolve(argv.root)
   }
 
+  if (argv.hmr == null) {
+    argv.hmr = true
```

Comment:
This ensures `config.hmr` is never undefined, so plugins can do `config.hmr ? thenThis : elseThat` instead of `config.hmr !== false`.